content/blog/end-to-end-easy-to-use-pipeline-for-training-a-model-on-medmnist-v2-using-hpe-machine-learning-development-environment-flask.md

```diff
@@ -0,0 +1,514 @@
+                    transforms.Normalize(mean=[0.5], std=[0.5]),
+                ]
+            )
+        else:
+            data_transform = transforms.Compose(
+                [transforms.ToTensor(), transforms.Normalize(mean=[0.5], std=[0.5])]
+            )
+
+        train_dataset = DataClass(
+            split="train",
+            transform=data_transform,
+            download=False,
+            as_rgb=True,
+            root=DATASET_ROOT,
+        )
+        train_loader = determined.pytorch.DataLoader(
+            dataset=train_dataset,
+            batch_size=self.context.get_per_slot_batch_size(),
+            shuffle=True,
+        )
+
+        return train_loader
+```
+
+In each function, we initialize and return the relevant PyTorch DataLoader object (e.g. `val_loader from build_validation_data_loader`, and `train_loader in build_training_data_loader`).  
+
+#### Step 1.4 Port Training and Evaluation Functions 
+
+Finally, we’ll port our training and evaluation functions to the following class functions by including the relevant steps to train and evaluate our model on one batch of data. Here, make sure to remove the for-loop iterating over epochs and include only the relevant code corresponding to one data batch. HPE Machine Learning Development Environment handles the training loop behind the scenes.  
+
+```python
+    def train_batch(
+        self, batch: TorchData, epoch_idx: int, batch_idx: int
+    ) -> Dict[str, Any]:
+        inputs, targets = batch
+        outputs = self.model(inputs)
+
+        if self.context.get_hparam("task") == "multi-label, binary-class":
+            targets = targets.to(torch.float32)
+            loss = self.criterion(outputs, targets)
+        else:
+            targets = torch.squeeze(targets, 1).long()
+            loss = self.criterion(outputs, targets)
+
+        self.context.backward(loss)
+        self.context.step_optimizer(self.optimizer)
+
+        return {"loss": loss}
+
+    def evaluate_batch(self, batch: TorchData) -> Dict[str, Any]:
+        inputs, targets = batch
+        outputs = self.model(inputs)
+
+        if self.context.get_hparam("task") == "multi-label, binary-class":
+            targets = targets.to(torch.float32)
+            loss = self.criterion(outputs, targets)
+            m = nn.Sigmoid()
+            outputs = m(outputs)
+        else:
+            targets = torch.squeeze(targets, 1).long()
+            loss = self.criterion(outputs, targets)
+            m = nn.Softmax(dim=1)
+            outputs = m(outputs)
+            targets = targets.float().resize_(len(targets), 1)
+
+        return {"test_loss": loss}
+```
+
+Notice that in all functions we’ve defined as part of MyMEDMnistTrial, we have removed all instances of manual device management, metric logging, and accuracy calculations. HPE Machine Learning Development Environment’s Trial APIs handle these automatically, given only the Trial definition and configuration file.  
```

Comment:
This one's a little tricky to reword to get the "royal 'we'" out of it. Give this a try:

"Ntice that in all functions defined as part of MyMEDMnistTrial, all instances of manual device management, metric logging, and accuracy calculations have been removed."